### PR TITLE
Rector-ize up to PHP 7.2

### PR DESCRIPTION
**Pull request by szepeviktor**
Really upgrade code.

Please consider using short array syntax. Some parts of WPCS is for visually (mentally?) impaired people.

From https://github.com/polylang/polylang/pull/1470#issuecomment-2063211826

### CODE REVIEW

**Comment by szepeviktor** [comment]
Rector cannot format code! It needs `phpcbf`.

File: admin/admin-notices.php [hunk 1]
```diff
@@ -177,17 +177,17 @@ public function hide_notice() {
 	public function display_notices() {
 		if ( current_user_can( 'manage_options' ) ) {
 			// Core notices
-			if ( defined( 'WOOCOMMERCE_VERSION' ) && ! defined( 'PLLWC_VERSION' ) && $this->can_display_notice( 'pllwc' ) && ! $this->is_dismissed( 'pllwc' ) ) {
+			if ( defined( 'WOOCOMMERCE_VERSION' ) && ! defined( 'PLLWC_VERSION' ) && $this->can_display_notice( 'pllwc' ) && ! static::is_dismissed('pllwc') ) {
```
Comment (szepeviktor) [inline admin/admin-notices.php:180]: e.g. this is a hard error 👀

File: rector.php [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    $rectorConfig->sets([
+        LevelSetList::UP_TO_PHP_72,
+/*
+        SetList::CODE_QUALITY,
+        SetList::DEAD_CODE,
+        SetList::CODING_STYLE,
+        SetList::TYPE_DECLARATION,
+        SetList::NAMING,
+        SetList::PRIVATIZATION,
+        SetList::EARLY_RETURN,
```
Comment (szepeviktor) [inline rector.php:30]: Rector can do Much More.

Comment (manooweb) [reply rector.php:30]: > Rector can do Much More. Please learn it if you love it.

I think we will love it 😁  Thanks @szepeviktor 🍺

**Comment by manooweb** [comment]
Hello @szepeviktor,

We're about to merge your work. Could you please solve the two conflicts before?

Thank you so much.

Regards.

**Comment by szepeviktor** [comment]
Done.

**Comment by manooweb** [comment]
> Done.

Thanks.

It's ok for me.
Could you also please run `phpcbf` as you said?

> Rector cannot format code! It needs `phpcbf`.

Otherwise we'll get `phpcs` errors 

```
FILE: C:\polylang\polylang\admin\view-translations-post.php
--------------------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
--------------------------------------------------------------------------------
 66 | ERROR | [x] Line indented incorrectly; expected at least 2 tabs, found 0
    |       |     (Generic.WhiteSpace.ScopeIndent.Incorrect)
--------------------------------------------------------------------------------
PHPCBF CAN FIX THE 1 MARKED SNIFF VIOLATIONS AUTOMATICALLY
--------------------------------------------------------------------------------

```

**Comment by szepeviktor** [comment]
Please run it instead of me.
I do not have development environment (because I do not write code).

**Comment by manooweb** [comment]
> Please run it instead of me. I do not have development environments (because I do not write code).

OK. So I will do another PR.
I saw that the default rector generated config file hasn't the same code as yours i.e. closer than its documentation.
I mean something like

```php
RectorConfig::configure()
    ->withPaths( 
       ....
    )
    ->withSets( 
       ...
    )
    ->withSkip( 
       ...
    ) 
```
What do you recommend?

**Comment by szepeviktor** [comment]
> What do you recommend?

I do not use Rector. The current configuration is all I can provide.

**Comment by manooweb** [comment]
Closed in favor of #1491. Thanks again @szepeviktor 😉👍

**Comment by szepeviktor** [comment]
You seem to be a programmer.

**Pull request closed without merging**